nulib/meadow
# Updates configuration and mix tasks to support v2 index

# Summary 
We need to be able to support v1 and v2 simultaneously.

# Specific Changes in this PR
- Adds v2 indexes to search config
- Creates v2 pipeline and script files
- Creates mix tasks to setup and teardown ingest pipelines, scripts, and # Version bump required by the PR
- Placeholders for the settings/mappings files for collection, file set and work v2 indexes.

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test
Run `mix meadow.elasticsearch.setup` to create the resources and `mix meadow.elasticsearch.teardown` to destroy the resources under your namespace/environment. Confirm in OpenSearch dashboard.

Also please let developers know if there are any special instructions to test this in the development environment. 

# :rocket: Deployment Notes

Mix setup task already applied in staging environment. To run in production, ssh into the prod container and run `mix meadow.elasticsearch.setup` from the `app` directory.

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
